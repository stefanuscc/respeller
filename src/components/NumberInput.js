import { useCallback } from "react"

export const NumberInput = (props) => {
    const { value, label, name, onChange, onLangChange, ...restProps } = props

    // Handle change number
    const handleChange = useCallback((event) => {
        onChange(event.target.value)
    }, [onChange])
    
    // Handle change language
    const handleLangChange = useCallback((event) => {
        onLangChange(event.target.value)
    }, [onLangChange])

    return (
        <div>
            <label htmlFor="number" className="block text-sm font-medium text-gray-700">
                {label}
            </label>
            <div className="mt-1 relative rounded-md shadow-sm">
                <div className="absolute inset-y-0 left-0 flex items-center">
                    <label htmlFor="language" className="sr-only">
                        Language
                    </label>
                    <select
                        id="lang"
                        name="lang"
                        onChange={handleLangChange}
                        className="focus:ring-indigo-500 focus:border-indigo-500 h-full py-0 pl-2 pr-2 border-transparent bg-transparent text-gray-500 sm:text-sm rounded-md"
                    >
                        <option value="en">EN</option>
                        <option value="id">ID</option>
                    </select>
                </div>
                <input
                    {...restProps}
                    id={name}
                    name={name}
                    type="number"
                    autoComplete={"off"}
                    required
                    value={value}
                    onChange={handleChange}
                    onKeyUp={handleChange}
                    className="appearance-none block w-full px-3 pl-16 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
                />
            </div>
        </div>
    )
}

export default NumberInput