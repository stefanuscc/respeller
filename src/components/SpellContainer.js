export const SpellContainer = (props) => {
    const { value, label, name, ...restProps } = props

    return (
        <div>
            <label htmlFor="spelling" className="block text-sm font-medium text-gray-700">
                {label}
            </label>
            <div className="mt-1">
                <textarea
                    {...restProps}
                    id={name}
                    name={name}
                    type={name}
                    value={value}
                    readOnly={true}
                    autoComplete="current-spelling"
                    required
                    className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
                ></textarea>
            </div>
        </div>
    )
}

export default SpellContainer

