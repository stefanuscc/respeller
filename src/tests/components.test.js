import { cleanup } from "@testing-library/react";
import { NumberInput } from "../components/NumberInput";
import { SpellContainer } from "../components/SpellContainer";

import renderer from "react-test-renderer"

afterEach(cleanup);

it("Renders proper loading number input", () => {
    const view = renderer.create(<NumberInput label="Number" name="number" value="30"/>);

    expect(view).toMatchSnapshot();
});

it("Renders proper loading spell container", () => {
    const view = renderer.create(<SpellContainer label="Spelling" name="spelling" value="Tiga Puluh"/>);

    expect(view).toMatchSnapshot();
});