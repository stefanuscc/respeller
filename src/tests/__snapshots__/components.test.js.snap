// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders proper loading number input 1`] = `
<div>
  <label
    className="block text-sm font-medium text-gray-700"
    htmlFor="number"
  >
    Number
  </label>
  <div
    className="mt-1 relative rounded-md shadow-sm"
  >
    <div
      className="absolute inset-y-0 left-0 flex items-center"
    >
      <label
        className="sr-only"
        htmlFor="language"
      >
        Language
      </label>
      <select
        className="focus:ring-indigo-500 focus:border-indigo-500 h-full py-0 pl-2 pr-2 border-transparent bg-transparent text-gray-500 sm:text-sm rounded-md"
        id="lang"
        name="lang"
        onChange={[Function]}
      >
        <option
          value="en"
        >
          EN
        </option>
        <option
          value="id"
        >
          ID
        </option>
      </select>
    </div>
    <input
      autoComplete="off"
      className="appearance-none block w-full px-3 pl-16 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
      id="number"
      name="number"
      onChange={[Function]}
      onKeyUp={[Function]}
      required={true}
      type="number"
      value="30"
    />
  </div>
</div>
`;

exports[`Renders proper loading spell container 1`] = `
<div>
  <label
    className="block text-sm font-medium text-gray-700"
    htmlFor="spelling"
  >
    Spelling
  </label>
  <div
    className="mt-1"
  >
    <textarea
      autoComplete="current-spelling"
      className="appearance-none block w-full px-3 py-2 border border-gray-300 rounded-md shadow-sm placeholder-gray-400 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"
      id="spelling"
      name="spelling"
      readOnly={true}
      required={true}
      type="spelling"
      value="Tiga Puluh"
    />
  </div>
</div>
`;
